thewalrus/_hafnian.py

```diff
@@ -178,6 +180,45 @@ def hafnian(
     )
 
 
+def hafnian_sparse(A, D = None):
+    r"""Returns the hafnian of a sparse symmetric matrix.
+    This pure python implementation is very slow on full matrices, but faster the sparser a matrix is.
+    As a rule of thumb, the crossover in runtime with respect to hafnian() happens around 50% sparsity.
+
+    Args:
+        A (array): the symmetric matrix of which we want to compute the hafnian
+        D (set): set of indices that identify a submatrix. If None (default) it computes
+        the hafnian of the whole matrix.
```

Comment:
![Screenshot from 2021-06-10 17-42-27](https://user-images.githubusercontent.com/24476053/121600940-407c2800-ca13-11eb-8426-6e9d8ea4305f.png)

```suggestion
            the hafnian of the whole matrix.
```